lib/git-util.js

```diff
@@ -0,0 +1,7 @@
+'use strict';
+const execa = require('execa');
+
+exports.hasUpstream = async () => {
+	const {stdout} = await execa('git', ['status', '--short', '--branch', '--porcelain=2']);
+	return stdout.split('\n').some(line => /^#\sbranch\.upstream\s[a-zA-Z0-9_\-/]+$/.test(line));
```

Comment:
```suggestion
	return stdout.split('\n').some(line => /^# branch\.upstream [\w\-/]+$/.test(line));
```

Reply:
Actually, couldn't it simply be ` return /^#\sbranch\.upstream\s[a-zA-Z0-9_\-/]+$/m.test(stdout)`?

Reply:
That seems to do the trick 👌🏻